examples/scene/prefab.rs

```diff
@@ -0,0 +1,205 @@
+use std::{fs::File, io::Write};
+use bevy::{ecs::schedule::ShouldRun, prelude::*, reflect::TypeRegistry, scene::SceneSpawnError, utils::HashMap};
+
+/// Basic Prefab support.
+/// To use the factory, spawn a component using Commands or World, and the Handle<DynamicScene> of the prefab.
+/// Then add to the factory queue. When the scene is loaded, 
+/// the components of the first entity listed in the file will be added to the entity.
+/// 
+/// LIMITATIONS
+///     Component data from the prefab won't be available immediately; the PrefabFactory system runs once per frame,
+///     and the actual scene must load before the scene is applied.
+///     
+///     Only the FIRST entity listed in the file will be inserted into the provided entity.
+///     All others will be spawned into the world as needed.
+///     Make sure to the other entities are parented somehow when writing the scene file so you can keep track of them!
+
+#[derive(Default)]
+pub struct PrefabFactory {
+    queue: HashMap<Entity, PrefabFactoryEntry>,
+}
+
+struct PrefabFactoryEntry {
+    handle: Handle<DynamicScene>,
+    overwrite_existing_components: bool
+}
+
+impl PrefabFactory {
+    pub fn add_to_queue(&mut self, entity: Entity, handle: Handle<DynamicScene>, overwrite_existing_components: bool) {
+        self.queue.insert(entity, PrefabFactoryEntry {
+            handle,
+            overwrite_existing_components,
+        });
+    }
+}
+
+#[derive(Reflect, Clone)]
+#[reflect(Component)]
```

Comment:
Do we not need `#[derive(Component)]` on these too? Or is that somehow implicitly done with the reflect macro?

Reply:
I'm still working on 0.5, so derive(Component) isn't a thing.

Reply:
We'll need it for this example though; since we're targeting main. Ditto with the .system dropping below.